src/Player/PlayerStrategies.cpp

```diff
@@ -0,0 +1,613 @@
+            if (country->get_armies() > biggest_army ){
+                biggest_army = country->get_armies();
+                strongest_neighbor = country;
+            }
+        }
+
+        if(strongest_neighbor == nullptr) {
+            Terminal::error("Strongest neighbor is null, this should never happen");
+            return answer;
+        }
+
+        int total_armies = strongest_neighbor->get_armies() + weakest_country->get_armies();
+        int total_armies_divided_by_two = total_armies / 2;
+        if (total_armies_divided_by_two * 2 == total_armies) {
+            strongest_neighbor->set_armies(total_armies_divided_by_two);
+            weakest_country->set_armies(total_armies_divided_by_two);
+        } else {
+            strongest_neighbor->set_armies(total_armies_divided_by_two+1);
+            weakest_country->set_armies(total_armies_divided_by_two);
+        }
+
+        answer.emplace_back(strongest_neighbor); // source
+        answer.emplace_back(weakest_country); //target
+        return answer;
+    }
+
+    void BenevolentPlayerStrategy::reinforce(int i) {
+        Terminal::debug("Performing reinforce from a benevolent player" + player->get_color());
+
+        vector<Board::Country *> owned_countries = player->get_countries();
+        //Find weakest country
+        int smallest_army = 99999;
+        Country * weakest_country = nullptr;
+        for (auto &country : owned_countries){
+            if (country->get_armies() < smallest_army){
+                weakest_country = country;
+                smallest_army = country->get_armies();
+            }
+        }
+
+        if(weakest_country == nullptr) {
+            Terminal::error("Weakest country is null, this should NEVER happen");
+            return;
+        }
+
+        weakest_country->set_armies(weakest_country->get_armies() + i);
+
+    }
+
+    BenevolentPlayerStrategy::~BenevolentPlayerStrategy() {
+
+    }
+
+    /**
+ * an aggressive computer player that focuses on attack (reinforces its strongest
+country, then always attack with it until it cannot attack anymore, then fortifies in order to maximize aggregation of
+forces in one country),
+ * @param i
+ */
+
+    bool AggressivePlayerStrategy::attack() {
+        bool has_won_a_battle = false;
+        //Find strongest country
+        int biggest_army = 0;
+        Country * strongest_country = nullptr;
+        for (auto &country : player->get_countries()){
+            if (country->get_armies() > biggest_army){
+                // Verify if the country has a neighbor that player owns and has army
+                for (auto &neighbor_country : * country->get_neighbors()){
+                    if (neighbor_country->get_owner() == player && neighbor_country->get_armies() > 1){
+                        strongest_country = country;
+                        biggest_army = country->get_armies();
+                        break;
+                    }
```

Comment:
That's a lot of question in a comment,

1) if it doesnt have a neighbor, it will be nullptr and will return false (which mean he will not attack)
2) if it only has a country, then it will be its strongest one
3) if it stays nullptr, that means there's no country to attack from. 

I understand your confusion since `neighbor_country->get_owner() == player` should be `neighbor_country->get_owner() != player` ( as in I should verify if the country as a neighbor that isn't his)